core/domain/prod_validators.py

```diff
@@ -5149,23 +5149,24 @@ def _validate_collections_are_marked_deleted(cls, item):
 
     @classmethod
     def _validate_activity_mapping_contains_only_allowed_keys(cls, item):
-        """Validates that activity_mappings keys are only from
+        """Validates that pseudonymizable_entity_mappings keys are only from
         the core.platform.models.NAMES enum.
 
         Args:
             item: PendingDeletionRequestModel. Pending deletion request model
                 to validate.
         """
         incorrect_keys = []
-        for key in item.activity_mappings.keys():
+        for key in item.pseudonymizable_entity_mappings.keys():
             if key not in [name for name in models.NAMES.__dict__]:
```

Comment:
Should this be stricter? (Not all entities are pseudonymizable, right?)

Reply:
Hmm, I think this should be fine, i.e. new additions to `models.NAMES` can be pseudonymizable and I don't think we should keep a subset of entities that can be pseudonymizable and add or remove stuff from there. The decision what should be done with the model should be driven by the `get_deletion_policy`.

Reply:
In that case, shouldn't this validation include a check on get_deletion_policy?

Reply:
Since we only have categories in here it is not possible to check the models directly, but thinking about it it won't hurt to have a list of models that can be pseudonymized as a double check.

Reply:
Done.